src/api/v2/handlers.rs

```diff
@@ -26,6 +31,33 @@ pub async fn subscriptions(
 	Ok(SubscriptionId { subscription_id })
 }
 
+pub async fn submit(
+	submitter: Arc<impl transactions::Submit>,
+	transaction: Transaction,
+) -> Result<TransactionHash, Error> {
+	fn base64_decode(data: String) -> Result<Vec<u8>, Error> {
+		general_purpose::STANDARD
+			.decode(data)
+			.map_err(|error| Error::bad_request(format!("Bad Request: {error}")))
+	}
+
+	if matches!(&transaction, Transaction::Data(_)) && !submitter.has_signer() {
+		return Err(Error::not_found());
+	};
+
+	let transaction = match transaction {
+		Transaction::Data(data) => transactions::Transaction::Data(base64_decode(data)?),
+		Transaction::Extrinsic(data) => transactions::Transaction::Extrinsic(base64_decode(data)?),
+	};
```

Comment:
Also there are 2 types of transactions here which is weird. You can squash them into one type and make it generic over container inside, or you can just have separate enum like `TransactionType`.

Reply:
Separate enum can imply different interface (e.g. `{"type":"da","data":"asdfaa88sfa"}` instaed, which is something Ill consider.

Reply:
We are not decided for an interface, so we can switch it right now with ease. So I think `TransactionType` sounds way better than having 2 same types

Reply:
This is fine at the moment since there are a lot unknowns, so lets keep it like this and continue discussion on the https://github.com/availproject/RFCs/discussions/3.

Reply:
RFC doesn't feature `extrinsic` for `/submit` though. For that, body will look like `{ "extrinsic": "adf1234" }`.

Let's move the discussion there, but I think RFC is not up-to-date if you've decided to have extrinsic variant also

Reply:
I've made a thread for RFC for discussion of the API: https://github.com/availproject/RFCs/discussions/3#discussioncomment-6876407